projects/client/src/app/pages/categories/pages/info/hooks/useLoadCategoryInfo.ts

```diff
@@ -1,7 +1,36 @@
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
+import { toast } from 'sonner';
 
-const useLoadCategoryInfo = () => {
-  useEffect(() => {}, []);
+import { Category } from '@/models/category.model';
+import { getCategoryById } from '@/services/category.service';
+
+type Props = {
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  setIsFound: React.Dispatch<React.SetStateAction<boolean>>;
+  id: string;
+};
+
+const useLoadCategoryInfo = ({ setIsLoading, setIsFound, id }: Props) => {
+  const [category, setCategory] = useState<Category>({} as Category);
+
+  useEffect(() => {
+    setIsLoading(true);
+    toast.promise(getCategoryById(id), {
+      loading: 'Loading category...',
+      success: (res) => {
+        setCategory(res);
+        setIsFound(true);
+        return `Category ${res.name} loaded successfully`;
+      },
+      error: () => {
+        setIsFound(false);
+        return 'Error Category not found';
```

Comment:
I think this is an error message displayed to customer, so should be enough: 'Category not found'

Reply:
In this cases is an error for the admin so i changed it to "Categories not found" instead